File: ClipboardZanager/Sources/ClipboardZanager/ComponentModel/CloudStorage/DropBoxTokenProvider.cs
using System.Collections.Generic;
using ClipboardZanager.Properties;
using ClipboardZanager.Shared.CloudStorage;
using ClipboardZanager.Shared.Logs;

namespace ClipboardZanager.ComponentModel.CloudStorage
{
    /// <summary>
    /// Provides a set of functions designed to provide tokens for a <see cref="DropBoxProvider"/>.
    /// </summary>
    internal sealed class DropBoxTokenProvider : ICloudTokenProvider
    {
        /// <inheritdoc/>
        public string GetToken(string tokenName)
        {
            switch (tokenName)
            {
                case "AppKey":
                    return Settings.Default.DropBoxAppKey;

                case "RedirectUri":
                    return Settings.Default.DropBoxRedirectUri;

                case "AccessToken":
                    return Settings.Default.DropBoxAccessToken;

                default:
                    Logger.Instance.Error(new KeyNotFoundException($"{tokenName} not found."));
                    return string.Empty;
            }
        }

        /// <inheritdoc/>
        public void SetToken(string tokenName, string value)
        {
            switch (tokenName)
            {
                case "AccessToken":
                    Settings.Default.DropBoxAccessToken = value;
                    break;

                default:
                    Logger.Instance.Error(new KeyNotFoundException($"{tokenName} not found."));
                    break;
            }

            Settings.Default.Save();
        }
    }
}

File: Android/app/src/main/java/com/etiennebaudoux/clipboardzanager/componentmodel/cloudstorage/providers/DropBoxTokenProvider.java
package com.etiennebaudoux.clipboardzanager.componentmodel.cloudstorage.providers;

import com.etiennebaudoux.clipboardzanager.App;
import com.etiennebaudoux.clipboardzanager.R;
import com.etiennebaudoux.clipboardzanager.componentmodel.cloudstorage.CloudTokenProvider;
import com.etiennebaudoux.clipboardzanager.componentmodel.core.CoreHelper;

/**
 * Provides a set of functions designed to provide tokens for a {@link DropBoxProvider}.
 */
public class DropBoxTokenProvider implements CloudTokenProvider {

    private static final String DropBoxAccessToken = "DropBoxAccessToken";

    @Override
    public String getToken(String tokenName) {
        switch (tokenName) {
            case "AppKey":
                return App.getContext().getString(R.string.DropBoxAppKey);

            case "AccessToken":
                return CoreHelper.getSetting(DropBoxAccessToken);

            default:
                return "";
        }
    }

    @Override
    public void setToken(String tokenName, String value) {
        switch (tokenName) {
            case "AccessToken":
                CoreHelper.setSetting(DropBoxAccessToken, value);
        }
    }
}
